yarn.lock

```diff
@@ -448,9 +448,9 @@
     "@babel/helper-plugin-utils" "^7.0.0"
 
 "@babel/plugin-transform-destructuring@^7.5.0":
-  version "7.5.0"
```

Comment:
```sh
git diff node_modules/@babel/plugin-transform-destructuring ../tmp/node_modules/@babel/plugin-transform-destructuring > patch.diff
```

```diff
diff --git a/node_modules/@babel/plugin-transform-destructuring/README.md b/../babel/node_modules/@babel/plugin-transform-destructuring/README.md
index 4c866eab2..92df503a1 100644
--- a/node_modules/@babel/plugin-transform-destructuring/README.md
+++ b/../babel/node_modules/@babel/plugin-transform-destructuring/README.md
@@ -1,3 +1,37 @@
+==================
+
+**This is a (hopefully temporary) fork of `@babel/plugin-transform-destructuring` that optimises array destructuring of `useState` and `useReducer` React Hooks.**
+
+**It only includes these two changes:**
+
+```diff
+@@ -1,6 +1,10 @@
+ import { declare } from '@babel/helper-plugin-utils';
+ import { types as t } from '@babel/core';
+
++const forcedLoose = ['useReducer', 'useState'].map(
++  name => new RegExp(`^_(?:React\\$)?${name}\\d*$`)
++);
++
+ export default declare((api, options) => {
+   api.assertVersion(7);
+```
+
+```diff
+@@ -129,7 +133,11 @@ export default declare((api, options) => {
+     }
+
+     toArray(node, count) {
+-      if (this.arrayOnlySpread || (t.isIdentifier(node) && this.arrays[node.name])) {
++      if (
++        this.arrayOnlySpread ||
++        (t.isIdentifier(node) &&
++          (this.arrays[node.name] || forcedLoose.some(regex => regex.test(node.name))))
++      ) {
+```
+
+==================
+
 # @babel/plugin-transform-destructuring
 
 > Compile ES2015 destructuring to ES5
diff --git a/node_modules/@babel/plugin-transform-destructuring/lib/index.js b/../babel/node_modules/@babel/plugin-transform-destructuring/lib/index.js
index 70e546780..8ef1de471 100644
--- a/node_modules/@babel/plugin-transform-destructuring/lib/index.js
+++ b/../babel/node_modules/@babel/plugin-transform-destructuring/lib/index.js
@@ -25,6 +25,10 @@ function _core() {
   return data;
 }
 
+const forcedLoose = ['useReducer', 'useState'].map(
+  name => new RegExp(`^_(?:React\\$)?${name}\\d*$`)
+);
+
 var _default = (0, _helperPluginUtils().declare)((api, options) => {
   api.assertVersion(7);
   const {
@@ -134,7 +138,11 @@ var _default = (0, _helperPluginUtils().declare)((api, options) => {
     }
 
     toArray(node, count) {
-      if (this.arrayOnlySpread || _core().types.isIdentifier(node) && this.arrays[node.name]) {
+      if (
+        this.arrayOnlySpread ||
+        (_core().types.isIdentifier(node) &&
+          (this.arrays[node.name] || forcedLoose.some(regex => regex.test(node.name))))
+      ) {
         return node;
       } else {
         return this.scope.toArray(node, count);
diff --git a/node_modules/@babel/plugin-transform-destructuring/package.json b/../babel/node_modules/@babel/plugin-transform-destructuring/package.json
index 91d8c52b3..05d4f771e 100644
--- a/node_modules/@babel/plugin-transform-destructuring/package.json
+++ b/../babel/node_modules/@babel/plugin-transform-destructuring/package.json
@@ -1,6 +1,6 @@
 {
-  "name": "@babel/plugin-transform-destructuring",
-  "version": "7.5.0",
+  "name": "@minh.nguyen/plugin-transform-destructuring",
+  "version": "7.5.2",
   "description": "Compile ES2015 destructuring to ES5",
   "repository": "https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-destructuring",
   "license": "MIT",

```

Reply:
I'm surprised this is even working in yarn. I would've expected it to regenerate it if you run `yarn`. I'll check if this is also working in yarn v2 but other than that this looks good now. 

Could you add a link to the github repro to the initial PR description?

Also curious what dependabot will do.

Reply:
> Could you add a link to the github repro to the initial PR description?

I'm guessing you meant `repo`, but I actually just did a `yarn add @babel/plugin-transform-destructuring` in an empty project, went into `node_modules` and edited `name` and `version` in `package.json` as well as `lib/index.js` and ran `npm publish`. Didn't have time to set up CI or anything like that.

> Also curious what dependabot will do.

To make it more reproducible, I could use Yarn resolutions and add a resolution for this.

Reply:
Updated the PR description with the diff as well as a link to the fork on npm.

Reply:
> To make it more reproducible, I could use Yarn resolutions and add a resolution for this.

I think dependabot doesn't deal with those very well but at least it helps readability.

I'm still very much opposed to this not being open source but since you did talk with @oliviertassinari privately he'll have to vouch for this change not being malicious.

Reply:
I invite you to review the package by installing the two side-by-side using `git diff` 🙂 I'll see if I can set up CI, GitHub etc on the weekend and then publish `7.5.3`, but no guarantees as I'm a little busy these days.

Reply:
Damn, just noticed some property called `gitHead` in `package.json` https://unpkg.com/browse/@minh.nguyen/plugin-transform-destructuring@7.5.2/package.json

After some googling, I think it's a property that Yarn and npm add when you install a package into `node_modules`. Do you think it's worth removing?